src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -405,36 +405,45 @@ public async Task ReinstateSubscriptionAsync(Guid organizationId)
         return new Tuple<bool, string>(success, paymentIntentClientSecret);
     }
 
-    private async Task ValidateSecretsManagerSeatsAndServiceAccount(OrganizationUpgrade upgrade, Organization organization, short newPlanSeats, Models.StaticStore.Plan newSecretsManagerPlan)
+    private async Task ValidateSecretsManagerSeatsAndServiceAccountAsync(OrganizationUpgrade upgrade, Organization organization,
+        Models.StaticStore.Plan newSecretsManagerPlan)
     {
         var newPlanSmSeats = (short)(newSecretsManagerPlan.BaseSeats +
                                      (newSecretsManagerPlan.HasAdditionalSeatsOption
                                          ? upgrade.AdditionalSmSeats
                                          : 0));
+        var occupiedSmSeats =
+            await _organizationUserRepository.GetOccupiedSmSeatCountByOrganizationIdAsync(organization.Id);
+
         if (!organization.SmSeats.HasValue || organization.SmSeats.Value > newPlanSmSeats)
         {
-            var occupiedSmSeats =
-                await _organizationUserRepository.GetOccupiedSmSeatCountByOrganizationIdAsync(organization.Id);
-            if (occupiedSmSeats > newPlanSeats)
+            if (occupiedSmSeats > newPlanSmSeats)
```

Comment:
Unnecessary nesting of the `if` statement. You don't actually need `occupiedSmSeats` below, so I think you can revert this part. (See my next comment as well)

Reply:
thanks